### ISSUE TITLE
Merge storybook docs and move to standalone package

### ISSUE THREAD

**Issue by zackkrida** [cc-archive/vocabulary-legacy#630]
## Updated Tasks

- [ ] Remove `fonts` storybook files; move those stories into the `packages/vocabulary` storybook
- [ ] Compose the vue storybook + the vocabulary storybook
- [x] Add/ CD script for `vue-vocabulary` (should run on release)
  - [x] Need to fix netlify config (see: https://www.netlify.com/blog/2019/10/09/launching-monorepo-support-for-netlify-sites/)

## Original Message:

All of the existing sites and storybook stories should be combined into one standalone site. That site should be located in `packages/docs`, which should have `private: true` in it's `package.json`.

This will allow us to decouple all storybook dependencies and configuration from our npm packages (fonts, vue-vocabulary, and vocabulary) and have a single source for all vocabulary information at https://cc-vocabulary.netlify.app/

The vue and fonts sites https://cc-fonts.netlify.app/ and https://cc-vue-vocabulary.netlify.app/ should be set to redirect to the main vocabulary site (I _think_ we can do this in Netlify).

**Comment by dhruvkb** [comment]
Or we could keep the storybooks separate which makes them more maintainable and use the [new Composition feature](https://medium.com/storybookjs/storybook-composition-af0da9084fba) to combine them into one.

**Comment by zackkrida** [comment]
I'm open to it. My original thinking was that the fonts package is iframe-ed into the main storybook anyway, and that it might be nice to colocate vue components next to the css versions, but I'm very open to other structures

**Comment by dhruvkb** [comment]
Composition allows for components inside different storybooks to be merged. It doesn't have to look like three separate storybooks located adjacent to each other.

**Comment by zackkrida** [comment]
I agree with what you said above. What file structure do you think makes the most sense? The two options I see are:

```shell
# Option A
packages/
├── docs
├── fonts
├──── stories
├── vocabulary
├──── stories
├── vue-vocabulary
└──── stories
```

```shell
# Option B
packages/
├── docs
├──── stories
├── fonts
├── vocabulary
└── vue-vocabulary
```

Basically, the idea of B is what I was referring to above. All stories in one storybook 'package' instead of the stories being in individual directories in each package.

Other than liking the idea of all stories being in one place, I don't feel too strongly about this, so happy to go in whatever direction you like.

**Comment by dhruvkb** [comment]
I really like option A, i.e. keeping stories with components, just to ensure that component changes are also accompanied by updates the docs. Again no strong preference either way so let's invite some more contributors to chime in on this.

@hugosolar @Dhruvi16 @chidexebere @nimishbongale any opinions on this?

**Comment by nimishbongale** [comment]
I just checked out the new Storybook Feature, and it looks great for our purposes! An easy way to combine existing Storybooks, thats for sure.

I'd really vouch for Option A.

Separating individual stories makes more sense imo, although I'm not entirely sure. Open for a discussion!

**Comment by zackkrida** [comment]
Thanks for the feedback @nimishbongale and @dhruvkb ! I like the new composition feature _a lot_. But there's a few reasons that I question us using it:

- `vue-vocabulary`: Every component in vue-vocabulary exists in vocabulary. I don't think its a good user experience to have a "Vocabulary" sidebar section that contains the "Button" component, and then a "Vue Vocabulary" sidebar section that has its own "Button" sub-section. I'd rather have a vue tab or setting pane inside the main "Button" section. So that would be an argument against composing a separate `vue-vocabulary` storybook.
- `fonts`: The fonts storybook is one page, and then a page to view each font face. One of these pages is _already iframe-ed_ into the main vocabulary storybook. I don't see a strong reason for this small amount of content to need its own storybook instance, especially if we're _already embedding it_, in a bit of a hack-y way.
- Maintenance: Using storybook composition we must maintain at least 3 separate `.storybook/` configurations that will likely be identical. We would need storybook js dependencies to be defined in the root of the project, so we don't have to maintain seperate versions of these dependencies in multiple packages (like we do now). We would either compose `vocabulary`, `vue-vocabulary`, and `fonts` into one master `docs` storybook package, _or_ compose `vue-vocabulary` and `fonts` into the `vocabulary` storybook instance. Lastly, the storybook composition api _uses iframes_ under the hod, and embeds sub-storybooks by url. So in development one would need to have 3-4 storybook processes running simultaneously, and in production we'd have to host and maintain 3-4 separate vocabulary sites, even if we encourage storybook users to only use one of them.

In conclusion, I'm leaning strongly _against_ the composition feature and towards _one_ storybook installation in a `docs/` package. I don't care where the actual _stories_ live, (that can be option A, as discussed).

Lots to think about here, so thanks for reading!

**Comment by dhruvkb** [comment]
All your points make sense, and I do think you're bringing up very valid points, especially about the `fonts` storybook. I can side with a single Storybook installation if we can work around these obstacles:
- Storybook for Vue supports DocsPage components like prop tables; Storybook for HTML does not. Combining them might not be trivial especially since many addons like knobs do not make sense in HTML as much as they do in Vue.
- It will be a massive task to compose the Webpack configuration of a unified Storybook that supports both SFC Vue components that are Babel-compiled and SCSS compiled to CSS.
- Adding new tabs to the top for Vue Vocabulary components might not be possible. I might be wrong but as far as I'm aware, only plugins are able to define new tabs.

**Comment by zackkrida** [comment]
Those are all really good points too. I could see an ideal world where all our stories are driven by vue in the future (currently, we have knobs on the main storybook, but it doesn't _do_ anything, for example) and then in our `docs` tab have a code block component that has a vue and vanilla tabs (shopify's polaris is a good example of this: https://polaris.shopify.com/components/forms/radio-button#navigation) so the stories embedded in the 'docs' tab can show both.

A good starting point though may be do eliminate `fonts` storybook files and compose the vocabulary + vue vocabulary storybooks until we transition into something more unified.

**Comment by dhruvkb** [comment]
Back in summer of '19, our docs were built using Styleguidist (instead of Storybook) because my primary reference was [Vue DS by Viljami Salminen](https://vueds.com/example/). All code samples shown on the site, like the [NavBar for example](https://vueds.com/example/#/Patterns/NavBar), have both Vue and HTML tabs.

**Comment by nimishbongale** [comment]
Seems like a good option to compose vocabulary and vue-vocabulary for now, as @zackkrida rightly mentioned. 

Then would that mean, @dhruvkb, that we could possibly find a way to just add in the VueJS snippets into the existing cc-vocabulary site?

I was really thinking about using [JSFiddle](https://jsfiddle.net/) here. The live run, auto code formatting & highlighting looks really good imo. 

Here's a snippet I prepared for our Cards Component. It renders in a markdown file just fine. I did try it out. 

`<iframe width="100%" height="300" src="//jsfiddle.net/nimishbongale/w9hv8z3q/2/embedded/html,result/" allowfullscreen="allowfullscreen" allowpaymentrequest frameborder="0"></iframe>`

It supports VueJS as well.

`<iframe width="100%" height="300" src="//jsfiddle.net/nimishbongale/cqy1b23a/embedded/js,html,result/" allowfullscreen="allowfullscreen" allowpaymentrequest frameborder="0"></iframe>`

The main functionality this adds is a 'live run' option, which we otherwise don't provide. Do let me know your thoughts on this!

**Comment by dhruvkb** [comment]
I'm not too keen on having our documents rendered inside of `iframes`. As for adding Vue snippets inside the Vocabulary storybook (which won't work), we should try rendering pure HTML snippets inside the Vue storybook (which, being valid Vue templates, should work).

**Comment by nimishbongale** [comment]
Okay. I guess we should discuss this over a meet or something! 🤔

**Comment by nimishbongale** [comment]
Post my discussion with @dhruvkb in our weekly GSOD meet, I realise that the use of JSFiddle isn't that necessary. Storybook makes it sufficiently easy to do it anyways, with an add-on. And as rightly pointed out by @dhruvkb, we would want vocabulary catering to not just developers, but a large audience of people.

The tabs switch between "HTML" and "Vue" does make intuitive sense to me as well 👍

**Comment by nimishbongale** [comment]
Just to get some clarity on this issue:

The current landing site exists in the `/docs` folder. If we are looking to compose vocabulary & vue-vocabulary we could just do with Vue Vocabulary composed inside of Vocabulary.

I'm not sure what the `packages/docs` folder would contain? I'm just thinking of just a docs revamp within the storybook for our 3 packages.

Or maybe we're looking to have just one storybook running in `packages/docs`? Yep, sounds good!

**Comment by possumbilities** [comment]
This project is moving towards being archived, and this item is no longer on the roadmap. Closing
